### ISSUE TITLE
[dde-dock] Applications should be restricted based on the Desktops/Workspaces

### ISSUE THREAD

**Issue by bopboop** [linuxdeepin/developer-center#256]
Let's assume that I have 2 workspaces and Chromium is pinned onto the dock:

On workspace 1, I have Chromium open.

When I go to workspace 2, Chromium should not be shown as open. It should allow me to open another chromium window on workspace 2.

It would be really nice if this enchantment was added!

**Comment by hualet** [comment]
I'm not sure if this is really necessary since we've provided a way to open new App instances quicky: by middle clicking the App icon.

**Comment by bopboop** [comment]
@hualet I still feel that it would be nice to have this toggle feature to isolate applications on the dock based on which desktop you are on. It helps reduce clutter. Like if I had firefox open on Desktop 1, it shouldn't be shown as open on Desktop 2 etc.

**Comment by hualet** [comment]
OK, need to discuss this with our PM.

**Comment by hualet** [comment]
@manateelazycat

**Comment by SalahAdDin** [comment]
This feature doesn't exist in `gnome`.

**Comment by brendonhc** [comment]
> This feature doesn't exist in `gnome`.

This feature exists in `gnome`!
https://askubuntu.com/questions/305962/setting-to-only-show-applications-of-current-workspace-in-launcher

> I'm not sure if this is really necessary since we've provided a way to open new App instances quicky: by middle clicking the App icon.

I believe that this feature would be sensational in deepin-dde, it's great to have the option of isolating or not isolating the different workspaces. It depends on the usage situation, it's very good, I used a lot in Ubuntu with gnome, then in the manjaro with kde and now unfortunately I'm missing a lot in Deepin! :(

When I talk about isolating a workspace, it means the dock will only signal how the current workspace apps open.

**Comment by thissayantan** [comment]
So anyone has any update on this issue?

**Comment by SalahAdDin** [comment]
Is this people still working in these repositories?

**Comment by kumarsiddy** [comment]
Yes This is a good feature, I also want this feature in deep In.  In gnome it used to be a feature and you could switch it on.

**Comment by Pismice** [comment]
How can 

> Yes This is a good feature, I also want this feature in deep In. In gnome it used to be a feature and you could switch it on.

Sorry but how can you turn it on in gnome ?
